File: 2020/day06.rb
require 'set'
input = File.read("/Users/hhh/JungleGym/advent_of_code/2020/inputs/day06.in").split("\n\n")

count = 0
counts = input.map do |group|
  q = Set.new
  copy = group.dup.gsub("\n", "")
  copy.chars.each do |c|
    q << c
  end
  q.size

  answers = Hash.new 0
  members = group.split("\n")
  members.each do |m|
    m.chars.each do |c|
      answers[c] += 1
    end
  end

  all_yes = 0

  answers.each do |question, responses|
    if responses == members.size
      all_yes += 1
    end
  end

  count += all_yes
end

res = counts.reduce(:+)

p "part 1: #{res}"
p "part 2: #{count}"

File: 2020/go/day06/day06.go
package main

import(
	"fmt"
	"io/ioutil"
	"strings"
)

func main() {
	input, err := ioutil.ReadFile("/Users/hhh/JungleGym/advent_of_code/2020/inputs/day06.in")
	if err != nil {
		panic(err)
	}
	groups := strings.Split(string(input), "\n\n")

	p1 := 0

	for _, g := range groups {
		unique := make(map[rune]bool)
		g = strings.ReplaceAll(g, "\n", "")
		for _, c := range g {
			unique[c] = true
		}
		p1 += len(unique)
	}
	fmt.Println(p1)
}
